test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+        threshold.muln(2),
+      );
+
+      if (poolType === "Xyk") {
+        const poolBalance = await getBalanceOfPool(
+          firstCurrency,
+          secondCurrency,
+        );
+        priceCalculationRetro = await calculate_sell_price_rpc(
+          poolBalance[0],
+          poolBalance[1],
+          threshold.muln(2),
+        );
+      } else {
+        priceCalculationRetro = sellPrice;
+      }
+
+      //we create reverse calculation to check that we receive another value
+      const sellPriceReverse = await calculateSellPriceByMarket(
+        liqId,
+        secondCurrency,
+        threshold.muln(2),
+      );
+
+      await signTx(
+        api,
+        Market.sellAsset(
+          liqId,
+          firstCurrency,
+          secondCurrency,
+          threshold.muln(2),
+        ),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const tokenAmount = await getTokensAccountInfo(
+        testUser.keyRingPair.address,
+        secondCurrency,
+      );
+
+      expect(sellPrice).bnEqual(priceCalculationRetro);
+      expect(sellPrice).not.bnEqual(sellPriceReverse);
+      expect(stringToBN(tokenAmount.free)).bnEqual(sellPrice);
+    });
+
+    test("Function calculateSellPriceWithImpact makes reliable calculations", async () => {
+      const sellPriceBefore = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      const sellPriceWithImpact = await calculateSellPriceWithImpact(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      await signTx(
+        api,
+        Market.sellAsset(liqId, firstCurrency, secondCurrency, threshold),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const sellPriceAfter = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      const diff = sellPriceAfter.sub(sellPriceWithImpact[1]);
+
+      expect(sellPriceBefore).bnEqual(sellPriceWithImpact[0]);
+      expect(diff).bnLt(sellPriceAfter.muln(3).divn(1000));
+    });
+
+    test("Function calculateBuyPriceByMarket works correctly", async () => {
+      const buyPrice = await calculateBuyPriceByMarket(
+        liqId,
+        secondCurrency,
+        threshold.muln(2),
+      );
+
+      if (poolType === "Xyk") {
+        const poolBalance = await getBalanceOfPool(
+          firstCurrency,
+          secondCurrency,
+        );
+        priceCalculationRetro = await calculate_buy_price_rpc(
+          poolBalance[0],
+          poolBalance[1],
+          threshold.muln(2),
+        );
+      } else {
+        priceCalculationRetro = buyPrice;
+      }
```

Comment:
duplicated. saw it in previous test.  plz refactor

Reply:
We check function calculateBuyPriceByMarket, we check calculateSellPriceByMarket in last one